js/test/Ice/ami/Client.ts

```diff
@@ -194,6 +194,8 @@ export class Client extends TestHelper {
                 test(r2.isSent());
                 test(r2.isCompleted());
 
+                await Promise.all(requests);
```

Comment:
The test previously assumed that once the last sent request completed, all prior requests had also completed. However, this assumption is incorrect, as responses may still be send in a different order due to server threading.